src/compiler/style/css-imports.ts

```diff
@@ -123,9 +123,10 @@ export const getCssImports = async (config: d.Config, compilerCtx: d.CompilerCtx
   styleText = stripCssComments(styleText);
 
   const dir = dirname(filePath);
-  const importeeExt = filePath.split('.').pop().toLowerCase();
+  const importedExt = filePath.split('.').pop().toLowerCase();
 
   let r: RegExpExecArray;
+  const IMPORT_RE = /(@import)\s+(url\()?\s?(.*?)\s?\)?([^;]*);?/gi;
```

Comment:
Moving this regex here doesn't change the code? If there was a problem with accessing `IMPORT_RE` , it would be sth like an "`IMPORT_RE` is not defined" runtime error, right? Can you explain how this solves your issue?

Reply:
It's not variable scoping issue, but rather that the [`RegExp.prototype.exec()` is stateful](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) when using the `/g` flag.
This causes subsequent iteration loops to have stale state and doesn't match import statements in the second file.
Moving the regex inside the function closure means that the state is fresh for each file, and successfully resolves all the imports.

Reply:
Ah yup makes sense, thanks for explaining! TIL ;)

I'm just a contributor, not a maintainer, but I'll approve this so they'll at least know someone else had a look already 🤓